app/assets/javascripts/test/libs/nml.spec.js

```diff
@@ -213,6 +230,27 @@ test("NML serializer should produce correct NMLs", t => {
   t.snapshot(serializedNml, { id: "nml" });
 });
 
+test("NML serializer should escape special characters and multilines", t => {
+  const state = update(initialState, {
+    tracing: {
+      description: { $set: "Multiline dataset\ndescription\nwith special &'<>\" chars." },
+      skeleton: {
+        trees: {
+          "1": {
+            comments: {
+              $push: [{ nodeId: 1, content: "Hello\"a'b<c>d&e\"f'g<h>i&j\nwith\nnew\nlines" }],
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const serializedNml = serializeToNml(state, state.tracing, state.tracing.skeleton, buildInfo);
+
+  t.snapshot(serializedNml, { id: "nml-special-chars" });
```

Comment:
Can you add another assertion here which makes sure serializedNml contains `Hello&quot;a&apos;b&lt;c&gt;d&amp;e&quot;f&apos;g&lt;h&gt;i&amp;j&#xa;with&#xa;new&#xa;lines`? Implicitly, this is also encoded in the snapshot, but I think the risk of losing that assertion by mistake is easier (e.g., when the serialization itself changes a lot) than if we would have a simple `t.true(serializedNml.indexOf(`...`) > -1`.